internal/checker/checker.go

```diff
@@ -15624,12 +15640,26 @@ func (c *Checker) getExportsOfModuleWorker(moduleSymbol *ast.Symbol) (exports as
 		}
 		return symbols
 	}
+	var originalModule *ast.Symbol
+	if moduleSymbol != nil {
+		if c.resolveSymbolEx(moduleSymbol.Exports[ast.InternalSymbolNameExportEquals], false /*dontResolveAlias*/) != nil {
+			originalModule = moduleSymbol
+		}
+	}
 	// A module defined by an 'export=' consists of one export that needs to be resolved
 	moduleSymbol = c.resolveExternalModuleSymbol(moduleSymbol, false /*dontResolveAlias*/)
 	exports = visit(moduleSymbol, nil, false)
 	if exports == nil {
 		exports = make(ast.SymbolTable)
 	}
+	// A CommonJS module defined by an 'export=' might also export JSTypeAliasDeclarations with
+	if originalModule != nil && len(originalModule.Exports) > 1 {
+		for _, symbol := range originalModule.Exports {
+			if symbol.Flags&ast.SymbolFlagsType != 0 && symbol.Name != ast.InternalSymbolNameExportEquals {
+				exports[symbol.Name] = symbol
```

Comment:
Since we're not merging symbols here, shouldn't we only promote the `JSTypeAliasDeclaration` symbols when they have unique names?

Reply:
Here, it would overwrite an existing symbol with that name.

Reply:
I think this is only possible in Typescript, because nothing in JS creates namespaces. You could get a conflict with `/** @typedef {1} x */ module.exports.x = 1` but this only works with `module.exports=`.

However, we may allow this in TS for 7.1 or later, so it's a good idea to add a check.